adapters/repos/db/vector/hnsw/config_update.go

```diff
@@ -112,36 +115,43 @@ func (h *hnsw) UpdateUserConfig(updated schema.VectorIndexConfig, callback func(
 		}
 	}
 
-	// ToDo: check atomic operation
+	if asyncEnabled() {
+		callback()
+		return nil
+	}
+
 	if !h.compressed.Load() {
 		// the compression will fire the callback once it's complete
-		h.turnOnCompression(parsed, callback)
+		return h.TurnOnCompression(callback)
 	} else {
 		// without a compression we need to fire the callback right away
 		callback()
+		return nil
 	}
+}
 
-	return nil
+func asyncEnabled() bool {
+	return os.Getenv("ASYNC_INDEXING") == "true"
 }
```

Comment:
`usecases/config/environment.go` contains methods used to fetch values of env variables. it can be used in here to provide consistent access to envs in terms of allowed values ("true", 1, "on", etc)

Reply:
Great point, I have exported the `Enabled` function and used it in there